src/providers/structure.ts

```diff
@@ -521,17 +520,49 @@ class StructureModel {
 }
 
 export class StructureTreeView {
+    private readonly extension: Extension
     private readonly _viewer: vscode.TreeView<Section | undefined>
     private readonly _treeDataProvider: SectionNodeProvider
     private _followCursor: boolean = true
 
 
     constructor(extension: Extension) {
+        this.extension = extension
         this._treeDataProvider = new SectionNodeProvider(extension)
         this._viewer = vscode.window.createTreeView('latex-workshop-structure', { treeDataProvider: this._treeDataProvider, showCollapseAll: true })
         vscode.commands.registerCommand('latex-workshop.structure-toggle-follow-cursor', () => {
            this._followCursor = ! this._followCursor
         })
+
+        vscode.workspace.onDidSaveTextDocument( (e: vscode.TextDocument) => {
+            if (extension.lwfs.isVirtualUri(e.uri)){
+                return
+            }
```

Comment:
Apart from this PR, we can make the structure view work on vitrualworkspaces since the feature does not depend on external commands like `pdflatex`, `chktex`, and `latexindent`.